Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -1421,10 +1350,33 @@ - (void)handleMediaTapGesture
         case OWSMessageCellType_ContactShare:
             [self.delegate didTapContactShareViewItem:self.viewItem];
             break;
-        case OWSMessageCellType_MediaAlbum:
+        case OWSMessageCellType_MediaAlbum: {
             OWSAssertDebug(self.bodyMediaView);
             OWSAssertDebug(self.viewItem.mediaAlbumItems.count > 0);
 
+            if (self.viewItem.mediaAlbumHasFailedAttachment) {
+                [self.delegate didTapFailedIncomingAttachment:self.viewItem];
+                return;
+            }
+
+            // TODO: We might be able to get rid of this.
```

Comment:
Just for my own understanding re: this TODO, do you mean to get rid of it and replace it with the "did tap on specific album cell" behavior?

Reply:
> Just for my own understanding re: this TODO, do you mean to get rid of it and replace it with the "did tap on specific album cell" behavior?

That's half of it. The other half of it is whether or not we need to special-case tapping on "lone" media v. "album" media.